language/move-vm/runtime/src/interpreter.rs

```diff
@@ -330,6 +307,19 @@ impl Interpreter {
         let return_values = result
             .result
             .map_err(|code| PartialVMError::new(StatusCode::ABORTED).with_sub_status(code))?;
+        // Paranoid check to protect us against incorrect native function implementations. A native function that
+        // returns a different number of values than its declared types will trigger this check
+        if return_values.len() != return_type_count {
```

Comment:
While we're at it, does it make sense to do some light type-checks of return parameters here as well?

Reply:
I was thinking primitive / struct types -- although do agree that it's not _super_ effective. For example, validating that a function which return `u64` does indeed return `u64`

Reply:
I added them in #509!